packages/amplication-git-utils/src/git/github.service.ts

```diff
@@ -306,31 +322,471 @@ export class GithubService implements GitProvider {
       auth: token,
     });
 
-    switch (pullRequestMode) {
-      case EnumPullRequestMode.Accumulative:
-        return new AccumulativePullRequest(
-          octokit,
-          owner,
-          repositoryName
-        ).createPullRequest(
-          pullRequestTitle,
-          pullRequestBody,
-          head,
-          files,
-          commit
-        );
-      default:
-        return new BasicPullRequest(
-          octokit,
-          owner,
-          repositoryName
-        ).createPullRequest(
-          pullRequestTitle,
-          pullRequestBody,
-          head,
-          files,
-          commit
-        );
+    const pr = await octokit.createPullRequest({
+      owner,
+      repo: repositoryName,
+      title: pullRequestTitle,
+      body: pullRequestBody,
+      head: branchName,
+      update: true,
+      changes: [
+        {
+          /* optional: if `files` is not passed, an empty commit is created instead */
+          files: files,
+          commit: commitMessage,
+        },
+      ],
+    });
+    return pr.data.html_url;
+  }
+
+  async createBranchIfNotExists({
+    owner,
+    repositoryName,
+    branchName,
+  }: CreateBranchIfNotExistsArgs): Promise<Branch> {
+    const { sha } = await this.getFirstDefaultBranchCommit(
+      owner,
+      repositoryName
+    );
+    const refs = await this.getBranch(owner, repositoryName, branchName);
```

Comment:
If the branch is not existing the function will throw an error

Reply:
I understand, but in this function which I deleted and handled differently, we are just swallowed the error and wrapped it in a boolean 
```
private async isBranchExist(branch: string): Promise<boolean> {
    try {
      const refs = await this.getBranch(branch);
      return Boolean(refs);
    } catch (error) {
      return false;
    }
  }
```